data/base/endpoint.php

```diff
@@ -104,28 +125,24 @@ protected function register() {
 	 * @param string $route
 	 * @param string $endpoint_class
 	 *
-	 * @return \Elementor\Data\Base\SubEndpoint
-	 * @throws \Exception
+	 * @return \Elementor\Data\Base\SubEndpoint|false
 	 */
-	protected function register_sub_endpoint( $route, $endpoint_class ) {
-		$endpoint_instance = new $endpoint_class( $route, $this );
+	public function register_sub_endpoint( $route, $endpoint_class ) {
+		$endpoint_instance = new $endpoint_class( $this, $route );
 
 		if ( ! ( $endpoint_instance instanceof SubEndpoint ) ) {
-			throw new \Exception( 'Invalid endpoint instance.' );
+			trigger_error( 'Invalid endpoint instance.' );
+			return false;
 		}
 
-		$endpoint_route = $route . '/' . $endpoint_instance->get_name();
+		$command = $endpoint_instance->get_full_command();
+		$format = $endpoint_instance::get_format();
```

Comment:
`get_format` should return `get_base_route` by default.

Reply:
Handled: https://github.com/elementor/elementor/pull/12820/commits/67cf8416bb7681a573ac13904e544eda3c4bf564